bridge-deploy/bridge_deploy/home.py

```diff
@@ -0,0 +1,118 @@
+    home_bridge_storage_contract: Contract = deploy_compiled_contract(
+        abi=eternal_storage_proxy_src["abi"],
+        bytecode=eternal_storage_proxy_src["bytecode"],
+        web3=web3,
+        transaction_options=transaction_options,
+        private_key=private_key,
+    )
+    increase_transaction_options_nonce(transaction_options)
+
+    # Deploy home bridge implementation
+    home_bridge_src = load_contract("HomeBridgeErcToNative")
+    home_bridge_contract: Contract = deploy_compiled_contract(
+        abi=home_bridge_src["abi"],
+        bytecode=home_bridge_src["bytecode"],
+        web3=web3,
+        transaction_options=transaction_options,
+        private_key=private_key,
+    )
+    increase_transaction_options_nonce(transaction_options)
+
+    # Connect home bridge proxy to implementation
+    home_bridge_storage_upgrade = home_bridge_storage_contract.functions.upgradeTo(
+        1, home_bridge_contract.address
+    )
+    send_function_call_transaction(
+        home_bridge_storage_upgrade,
+        web3=web3,
+        transaction_options=transaction_options,
+        private_key=private_key,
+    )
+    increase_transaction_options_nonce(transaction_options)
+
+    # Use proxy from now on
+    home_bridge_contract = web3.eth.contract(
+        abi=home_bridge_src["abi"], address=home_bridge_storage_contract.address
+    )
+
+    # TODO: Initialize bridge
+
+    # TODO: transfer ownership
+
+    contracts = DeployedHomeBridgeResult(home_bridge_contract, latest_block.number)
+
+    return contracts
+
+
+def initialize_home_bridge_contract(
+    *,
+    web3,
+    transaction_options: Dict = None,
+    home_bridge_contract,
+    validator_contract_address,
+    home_daily_limit,
+    home_max_per_tx,
+    home_min_per_tx,
+    home_gas_price,
+    required_block_confirmations,
+    block_reward_address,
+    foreign_daily_limit,
+    foreign_max_per_tx,
+    home_bridge_owner,
+    private_key=None,
+):
+    if transaction_options is None:
+        transaction_options = {}
+
+    home_bridge_contract_initialize = home_bridge_contract.functions.initialize(
+        validator_contract_address,
+        home_daily_limit,  # 30000000000000000000000000,  # HOME_DAILY_LIMIT in WEI,
+        home_max_per_tx,  # 1500000000000000000000000,  # HOME_MAX_AMOUNT_PER_TX in WEI,
+        home_min_per_tx,  # 500000000000000000,  # HOME_MIN_AMOUNT_PER_TX in WEI,
+        home_gas_price,  # 1000000000,  # HOME_GAS_PRICE in WEI,
+        required_block_confirmations,  # 4,  # HOME_REQUIRED_BLOCK_CONFIRMATIONS,
+        block_reward_address,  # "0x0000000000000000000000000000000000000000",  # BLOCK_REWARD_ADDRESS,
+        foreign_daily_limit,  # 15000000000000000000000000,  # FOREIGN_DAILY_LIMIT,
+        foreign_max_per_tx,  # 750000000000000000000000,  # FOREIGN_MAX_AMOUNT_PER_TX,
+        home_bridge_owner,  # "0x0000000000000000000000000000000000000000",  # HOME_BRIDGE_OWNER
```

Comment:
see above, I think this should be set to the private key's address and then in another transaction to 0.

Reply:
Initializing with deployer for now.

Reply:
Changed to initializing with address(1), as changing the owner to address(0) later seems to be forbidden as well.